Support for PubSub Lock on AWS Elasticache Valkey

kistlers opened the pull request:
This is an attempt to make issue #10471 work without support for `SPUBLISH`.
I append the `lockKey` to the `channelUnlockKey`. This allows us to control the cluster slot cache this operation is hashed into, by including a hash tag (`{...}`) in the `lockKey`. Refer to the issue description for an explanation of why this matters.

As a consequence, I had to change the `ChannelTopic` to a `PatternTopic` to keep receiving subscription events for the PubSub lock.

artembilan reviewed (changes requested):
Please, add DCO into your commit message before we start reviewing: https://spring.io/blog/2025/01/06/hello-dco-goodbye-cla-simplifying-contributions-to-spring.
Thanks

kistlers committed:
GH-10471: add the lockKey to the unlockChannelKey to control cluster slot hashing

As a consequence, change the ChannelTopic to a PatternTopic to keep receiving subscription events for PubSub lock

Signed-off-by: Severin Kistler <kistlerseverin@gmail.com>

kistlers committed:
GH-10471: pass unLockChannelKey as `KEYS[2]` instead of `ARGV[2]` as values used as keys should be passed as such

Signed-off-by: Severin Kistler <kistlerseverin@gmail.com>

artembilan reviewed (changes requested):
This looks more closer to what I won't worry about 😄 .

If you confirm that this works with Valkey, please, add your name to the `@author` list of the changed class.

Please, also rebase your branch (and re-push to PR) to the latest `main` as I've just pushed the fix Redis tests according to the latest Spring Data Redis.

Thanks
